advisors/spring-ai-advisors-vector-store/src/main/java/org/springframework/ai/chat/client/advisor/vectorstore/VectorStoreChatMemoryAdvisor.java

```diff
@@ -173,28 +206,103 @@ else if (message instanceof AssistantMessage assistantMessage) {
 		return docs;
 	}
 
-	public static class Builder extends AbstractChatMemoryAdvisor.AbstractBuilder<VectorStore> {
+	/**
+	 * Builder for VectorStoreChatMemoryAdvisor.
+	 */
+	public static class Builder {
 
 		private PromptTemplate systemPromptTemplate = DEFAULT_SYSTEM_PROMPT_TEMPLATE;
 
-		protected Builder(VectorStore chatMemory) {
-			super(chatMemory);
+		private Integer chatMemoryRetrieveSize = DEFAULT_CHAT_MEMORY_RESPONSE_SIZE;
+
+		private String conversationId = ChatMemory.DEFAULT_CONVERSATION_ID;
+
+		private Scheduler scheduler;
+
+		private int order = Advisor.DEFAULT_CHAT_MEMORY_PRECEDENCE_ORDER;
+
+		private VectorStore vectorStore;
+
+		/**
+		 * Creates a new builder instance.
+		 * @param vectorStore the vector store to use
+		 */
+		protected Builder(VectorStore vectorStore) {
+			this.vectorStore = vectorStore;
+		}
+
+		/**
+		 * Set the system prompt template.
+		 * @param systemPromptTemplate the system prompt template
+		 * @return this builder
+		 */
+		public Builder systemPromptTemplate(PromptTemplate systemPromptTemplate) {
+			this.systemPromptTemplate = systemPromptTemplate;
+			return this;
 		}
 
+		/**
+		 * Set the system text advice.
+		 * @param systemTextAdvise the system text advice
+		 * @return this builder
+		 */
 		public Builder systemTextAdvise(String systemTextAdvise) {
 			this.systemPromptTemplate = new PromptTemplate(systemTextAdvise);
 			return this;
 		}
 
-		public Builder systemPromptTemplate(PromptTemplate systemPromptTemplate) {
-			this.systemPromptTemplate = systemPromptTemplate;
+		/**
+		 * Set the chat memory retrieve size.
+		 * @param chatMemoryRetrieveSize the chat memory retrieve size
+		 * @return this builder
+		 */
+		public Builder chatMemoryRetrieveSize(int chatMemoryRetrieveSize) {
+			this.chatMemoryRetrieveSize = chatMemoryRetrieveSize;
+			return this;
+		}
+
+		/**
+		 * Set the conversation id.
+		 * @param conversationId the conversation id
+		 * @return the builder
+		 */
+		public Builder conversationId(String conversationId) {
+			this.conversationId = conversationId;
+			return this;
+		}
+
+		/**
+		 * Set whether to protect from blocking.
+		 * @param protectFromBlocking whether to protect from blocking
+		 * @return the builder
+		 */
+		public Builder protectFromBlocking(boolean protectFromBlocking) {
```

Comment:
I would remove this in favour of `scheduler()`

Reply:
same comment as before.